lib/ace/ext/papeeria/tex_completer.coffee

```diff
@@ -149,12 +160,41 @@ define((require, exports, module) ->
       else
         callback(null, @cache)
 
+  ###
+  # Show popup if token.type at current pos is one of allowedTypes[i]
+  # @ (editor) --> editor
+  # @ (list of strings) -- allowedTypes
+  ###
+  showPopupIfTokenIsOneOfTypes = (editor, allowedTypes) ->
+    if editor.completer?
+      pos = editor.getCursorPosition()
+      session = editor.getSession()
+      token = editor.session.getTokenAt(pos.row, pos.column)
+
+      if token?
+        for type in allowedTypes
+          if LatexParsingContext.isType(token, type)
+            editor.completer.showPopup(editor)
+
   class TexCompleter
       constructor: ->
-        @refGetter = new ReferenceGetter()
-      @init: (editor) ->  init(editor,  {win: "enter", mac: "enter"})
-
+        @refGetter = new PropertyGetter(processReferenceJson)
+        @citeGetter = new PropertyGetter(processCitationJson)
+      @init: (editor) ->
+        init(editor,  {win: "enter", mac: "enter"})
+
+        # Auto showing popup in ref, cite and other context
+        editor.commands.on('afterExec', (event) ->
```

Comment:
Grammar nazi mode: _it doesn't work_. You either use present simple or present continuous, not both

Code reader mode: _it doesn't work_ is not true -- it does work fine when you place cursor inside already existing `ref{}`, and that's exacty what reader expects to see when he reads this code. However, there is one specific (but very important) case when it really doesn't work, and this case should be documented here, so that it left no places for silly questions.